### PR TITLE
Backport to 1.1.1.: Fix err checking and mem leaks of BIO_set_conn_port and BIO_set_conn_address

### PR DESCRIPTION
**Pull request by DDvO**
This is the backport mentioned in #12034 that I had intended,
according to a suggestion by @bernd-edlinger: https://github.com/openssl/openssl/pull/11986#discussion_r434608708 
who had discovered these issues in `bss_conn.c` and reviewed my fix there.

On this occasion the PR also replaces a remnant `BUF_strdup()` call by `OPENSSL_strdup()` in `bss_acpt.c`.

### CODE REVIEW

File: crypto/bio/bss_acpt.c [hunk 1]
```diff
@@ -434,7 +434,7 @@ static long acpt_ctrl(BIO *b, int cmd, long num, void *ptr)
                 b->init = 1;
             } else if (num == 1) {
                 OPENSSL_free(data->param_serv);
-                data->param_serv = BUF_strdup(ptr);
+                data->param_serv = OPENSSL_strdup(ptr);
```
Comment (bernd-edlinger) [inline crypto/bio/bss_acpt.c:437]: I have not seen this in the other PR, but also here the error handling is missing.

Comment (DDvO) [reply crypto/bio/bss_acpt.c]: > I have not seen this in the other PR,

`BUF_strdup()` is not any more used in current master.

> but also here the error handling is missing.

Good point - corrected, also in #11986.

File: test/handshake_helper.c [hunk 2]
```diff
@@ -637,7 +637,8 @@ static int configure_handshake_ctx(SSL_CTX *server_ctx, SSL_CTX *server2_ctx,
     }
     if (extra->client.alpn_protocols != NULL) {
         unsigned char *alpn_protos = NULL;
-        size_t alpn_protos_len;
+        size_t alpn_protos_len = 0;
```
Comment (bernd-edlinger) [inline test/handshake_helper.c:640]: really that should be another PR, and probably also for master

**Comment by DDvO** [comment]
One Travis build failed: https://travis-ci.org/github/openssl/openssl/jobs/694560274
due to an unrelated issue, namely a gcc false positive warning on potentially uninitialized `alpn_protos_len` variable in a `TEST_true()` macro call in `test/handshake_helper.c`.
I've added a commit initializing the variable such that the (here wrong) warning is prevented.

File: test/handshake_helper.c (hunk 2 above)
Comment (DDvO) [reply test/handshake_helper.c]: Ok, I've removed it here and will add separate PR(s). 
The one Travis failure resulting from that issue can be ignored for this PR.
Can you please re-confirm?

**Comment by DDvO** [comment]
I still need an OTC approval here.
The Travis failure is unrelated and handled in #12041.

**Comment by DDvO** [comment]
Pushed to `OpenSSL_1_1_1-stable` - thanks @bernd-edlinger and @t8m
